src/sage/rings/semirings/tropical_variety.py

```diff
@@ -0,0 +1,875 @@
+        sphinx_plot(p1.tropical_variety().plot())
+
+    A slightly different result will be obtained if we use min-plus algebra
+    for the base tropical semiring::
+
+        sage: T = TropicalSemiring(QQ, use_min=True)
+        sage: R.<x,y> = PolynomialRing(T)
+        sage: p1 = R(1)*x + x*y + R(0)
+        sage: tv = p1.tropical_variety(); tv
+        Tropical curve of 0*x*y + 1*x + 0
+        sage: tv.components()
+        [[(t1, 1), [t1 <= -1], 1], [(-1, t1), [t1 >= 1], 1], [(-t1, t1), [t1 <= 1], 1]]
+        sage: tv.plot()
+        Graphics object consisting of 3 graphics primitives
+        
+    .. PLOT::
+        :width: 300 px
+
+        T = TropicalSemiring(QQ, use_min=True)
+        R = PolynomialRing(T, ('x,y'))
+        x, y = R.gen(), R.gen(1)
+        p1 = R(1)*x + x*y + R(0)
+        sphinx_plot(p1.tropical_variety().plot())
+
+    Tropical variety can consist of multiple components with varying orders::
+
+        sage: T = TropicalSemiring(QQ, use_min=False)
+        sage: R.<x,y> = PolynomialRing(T)
+        sage: p1 = R(7) + T(4)*x + y + R(4)*x*y + R(3)*y^2 + R(-3)*x^2
+        sage: tv = p1.tropical_variety(); tv
+        Tropical curve of (-3)*x^2 + 4*x*y + 3*y^2 + 4*x + 0*y + 7
+        sage: tv.components()
+        [[(3, t1), [t1 <= 0], 1],
+        [(-t1 + 3, t1), [0 <= t1, t1 <= 2], 1],
+        [(t1, 2), [t1 <= 1], 2],
+        [(t1, 0), [3 <= t1, t1 <= 7], 1],
+        [(7, t1), [t1 <= 0], 1],
+        [(t1 - 1, t1), [2 <= t1], 1],
+        [(t1 + 7, t1), [0 <= t1], 1]]
+        sage: tv.plot()
+        Graphics object consisting of 8 graphics primitives
+        
+    .. PLOT::
+        :width: 300 px
+
+        T = TropicalSemiring(QQ, use_min=False)
+        R = PolynomialRing(T, ('x,y'))
+        x, y = R.gen(), R.gen(1)
+        p1 = R(7) + T(4)*x + y + R(4)*x*y + R(3)*y**2 + R(-3)*x**2
+        sphinx_plot(p1.tropical_variety().plot())
+
+    If the tropical polynomial have `n>2` variables, then the result will be 
+    a tropical hypersurface embedded in a real space `\RR^n`::
+
+        sage: T = TropicalSemiring(QQ)
+        sage: R.<a,x,y,z> = PolynomialRing(T)
+        sage: p1 = x*y + R(-1/2)*x*z + R(4)*z^2 + a*x
+        sage: tv = p1.tropical_variety(); tv
+        Tropical hypersurface of 0*a*x + 0*x*y + (-1/2)*x*z + 4*z^2
+        sage: tv.components()
+        [[(t1, t2, t3 - 1/2, t3), [t2 - 9/2 <= t3, t3 <= t1 + 1/2, t2 - 5 <= t1], 1],
+        [(t1, 2*t2 - t3 + 4, t3, t2), [t3 + 1/2 <= t2, t3 <= t1], 1],
+        [(t1, t2, t1, t3), [max(t1 + 1/2, 1/2*t1 + 1/2*t2 - 2) <= t3], 1],
+        [(t1, t2 + 9/2, t3, t2), [t2 <= min(t3 + 1/2, t1 + 1/2)], 1],
+        [(t1 - 1/2, t2, t3, t1), [t2 - 9/2 <= t1, t1 <= t3 + 1/2, t2 - 5 <= t3], 1],
+        [(2*t1 - t2 + 4, t2, t3, t1), [t1 <= min(1/2*t2 + 1/2*t3 - 2, t2 - 9/2)], 1]] 
+    """
+    def __init__(self, poly):
+        r"""
+        Initialize ``self``.
+
+        EXAMPLES:
+
+        Tropical variety in two, three, and higher dimensions::
```

Comment:
These examples/tests should just be a `TestSuite(tv).run()` with keeping the current `TESTS::` block too. The rests of these should be covered by other methods.